woocommerce/woocommerce
# Pre-creating a brand to assign it to the coupon afterward.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We have a situation when one of the brand tests depends on the brand created by the previous one. Refactoring brand related tests into a self-contained ones. 

Closes #53371  .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. `pnpm env:start`
2. `pnpm test:e2e create-restricted-coupons.spec.js` 
3. Observe all the tests pass and especially the seventh one ` Restricted coupon management › can create new excludeProductBrands coupon`

<img width="1376" alt="Cursor_and_wordpress_–_README_md" src="https://github.com/user-attachments/assets/91148d79-a792-42b0-a4f1-586f37ec385d">

4. `pnpm test:e2e page-loads.spec.js`
5. Observe all the tests pass as well and, especially, ours ` WooCommerce Page Load > Load Marketing sub pages › Can load Coupons`

<img width="1181" alt="Cursor_and_wordpress_–_README_md" src="https://github.com/user-attachments/assets/adad4bff-af66-42bc-bd67-b93dd1fbaed1">

### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

This PR fixes the development-related issue.

</details>
